onnxruntime/test/python/onnxruntime_test_ort_trainer.py

```diff
@@ -261,8 +274,20 @@ def get_model(self):
         model_desc = MNISTWrapper.mnist_model_description()
         return model, model_desc
 
-    def get_trainer(self, model, model_desc, device, onnx_opset_ver=12, frozen_weights=[]):
-        return ORTTrainer(model, MNISTWrapper.my_loss, model_desc, "SGDOptimizer", None, IODescription('Learning_Rate', [1, ],
+    def get_model_with_internal_loss(self):
+        input_size = 784
+        hidden_size = 500
+        num_classes = 10
+
+        # warning: changes the pytorch random generator state
+        model = MNISTWrapper.NeuralNetWithLoss(input_size, hidden_size, num_classes)
+        model_desc = MNISTWrapper.mnist_model_description()
+        return model, model_desc
+
+    def get_trainer(self, model, model_desc, device, onnx_opset_ver=12, frozen_weights=[],
+                    internal_loss_fn=False):
+        loss_fn = MNISTWrapper.my_loss if not internal_loss_fn else None
+        return ORTTrainer(model, loss_fn, model_desc, "SGDOptimizer", None, IODescription('Learning_Rate', [1, ],
```

Comment:
for new test cases, it is better to use internal lr generator (get_lr_this_step) and if needed (loss scaler generator). This is the design will choose. So that once the new design is implemented there is no extra work.

Reply:
updated

Reply:
nip: do we still need to pass in IODescription('Learning_Rate', [1, ], if there is a get_lr_this_step?